src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/BaseTest.cs

```diff
@@ -75,6 +79,67 @@ public void BeforeAllTests ()
 				} catch (Exception ex) {
 					Console.Error.WriteLine ("Failed to determine whether there is Android target emulator or not: " + ex);
 				}
+
+				CreateBinutilsSymlink ();
+			}
+
+			void CreateBinutilsSymlink ()
+			{
+				string hostNdkDir = GetHostBinutilsDir ();
+				string xabtDir = Path.GetDirectoryName (typeof (BuildApk).Assembly.Location);
+				string symlinkDir = Path.Combine (xabtDir, hostNdkDir);
+
+				// If directory exists then it's either a valid symlink or a real dir, don't touch anything in such instance
+				if (Directory.Exists (symlinkDir)) {
+					if (!SymbolicLink.IsPathSymlink (symlinkDir)) {
+						Console.Error.WriteLine ($"Warning: a directory exists where a symbolic link was expected, {symlinkDir}");
+					}
+					return;
+				}
+
+				// If a file exists and is a symlink, then it's probably a dangling link - recreate it
+				if (File.Exists (symlinkDir)) {
+					if (!SymbolicLink.IsPathSymlink (symlinkDir)) {
+						throw new InvalidOperationException ($"File named '{symlinkDir}' exists where a symbolic link to a directory should be.");
+					}
+
+					File.Delete (symlinkDir);
+				}
+
+				// We first try the in-tree location...
+				string binutilsDirFullPath = Path.Combine (XABuildPaths.PrefixDirectory, "lib", "xamarin.android", "xbuild", "Xamarin", "Android", hostNdkDir);
+				if (!Directory.Exists (binutilsDirFullPath)) {
+					// ...since it failed, we'll try to find the system location
+					binutilsDirFullPath = Path.Combine (TestEnvironment.MonoAndroidToolsDirectory, hostNdkDir);
+					if (!Directory.Exists (binutilsDirFullPath)) {
+						throw new InvalidOperationException ($"Host NDK directory does not exist, {binutilsDirFullPath}");
+					}
+				}
```

Comment:
If we do restore this, you can use Setup.AndroidMSBuildDirectory here instead (it will do the same local build tree check first): https://github.com/xamarin/xamarin-android/blob/75a0743ed078d0cbdb07fc05ff91165147520ab2/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/BaseTest.cs#L40

```suggestion
				string binutilsDirFullPath = Path.Combine (AndroidMSBuildDirectory, hostNdkDir);
				if (!Directory.Exists (binutilsDirFullPath)) {
					throw new InvalidOperationException ($"Host NDK directory does not exist, {binutilsDirFullPath}");
				}
```